vault/external_tests/seal_binary/seal_sighup_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build !enterprise
+
+package seal_binary
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestSealReloadSIGHUP(t *testing.T) {
+	binary := os.Getenv("VAULT_BINARY")
+	if binary == "" {
+		t.Skip("only running docker test with $VAULT_BINARY present")
+	}
+
+	transitContainer, transitConfig, err := createTransitTestContainer("hashicorp/vault", "latest", 2)
+	if err != nil {
+		t.Fatalf("error creating vault container: %s", err)
+	}
+	defer transitContainer.Cleanup()
+
+	firstTransitKeyConfig := fmt.Sprintf(transitParameters,
+		transitConfig.Address,
+		transitConfig.Token,
+		transitConfig.MountPaths[0],
+		transitConfig.KeyNames[0],
+		"transit-seal-1",
+	)
+
+	secondTransitKeyConfig := fmt.Sprintf(transitParameters,
+		transitConfig.Address,
+		transitConfig.Token,
+		transitConfig.MountPaths[1],
+		transitConfig.KeyNames[1],
+		"transit-seal-2",
+	)
+
+	testCases := map[string]struct {
+		sealStanzas       []string
+		expectedSealTypes []string
+	}{
+		"migrate transit to transit": {
+			sealStanzas: []string{
+				fmt.Sprintf(transitStanza, firstTransitKeyConfig, 1, "false"),
+				fmt.Sprintf(transitStanza, firstTransitKeyConfig, 2, "true") + "," +
+					fmt.Sprintf(transitStanza, secondTransitKeyConfig, 1, "false"),
+				fmt.Sprintf(transitStanza, secondTransitKeyConfig, 1, "false"),
+			},
+			expectedSealTypes: []string{
+				"transit",
+				"transit",
+				"transit",
+			},
+		},
+		"migrate shamir to transit fails": {
+			sealStanzas: []string{
+				"",
+				fmt.Sprintf(transitStanza, firstTransitKeyConfig, 1, "false"),
+			},
+			expectedSealTypes: []string{
+				"shamir",
+				"shamir",
+			},
+		},
+		"migrate transit to shamir fails": {
+			sealStanzas: []string{
+				fmt.Sprintf(transitStanza, firstTransitKeyConfig, 1, "false"),
+				"",
+			},
+			expectedSealTypes: []string{
+				"transit",
+				"transit",
+			},
+		},
+		"replacing seal fails": {
+			sealStanzas: []string{
+				fmt.Sprintf(transitStanza, firstTransitKeyConfig, 1, "false"),
+				fmt.Sprintf(transitStanza, secondTransitKeyConfig, 1, "false"),
+			},
+			expectedSealTypes: []string{
+				"transit",
+				"transit",
+			},
+		},
+		"more than one seal fails": {
+			sealStanzas: []string{
+				fmt.Sprintf(transitStanza, firstTransitKeyConfig, 1, "false"),
+				fmt.Sprintf(transitStanza, firstTransitKeyConfig, 1, "false") + "," +
+					fmt.Sprintf(transitStanza, secondTransitKeyConfig, 2, "false"),
```

Comment:
I am curious about this test case. Why doesn't going from autoseal to multiseal work?

Reply:
These tests only run for non-enterprise builds, so configuring multiple enabled seals will fail